driver_test.go

```diff
@@ -149,16 +150,84 @@ func runTests(t *testing.T, dsn string, tests ...func(dbt *DBTest)) {
 	}
 
 	for _, test := range tests {
+		test := test
 		t.Run("default", func(t *testing.T) {
 			dbt := &DBTest{t, db}
+			t.Cleanup(func() {
+				dbt.db.Exec("DROP TABLE IF EXISTS test")
+			})
 			test(dbt)
 			dbt.db.Exec("DROP TABLE IF EXISTS test")
 		})
 		if db2 != nil {
 			t.Run("interpolateParams", func(t *testing.T) {
 				dbt2 := &DBTest{t, db2}
+				t.Cleanup(func() {
+					dbt2.db.Exec("DROP TABLE IF EXISTS test")
+				})
 				test(dbt2)
-				dbt2.db.Exec("DROP TABLE IF EXISTS test")
+			})
+		}
+	}
+}
+
+func runTestsParallel(t *testing.T, dsn string, tests ...func(dbt *DBTest, tableName string)) {
+	if !available {
+		t.Skipf("MySQL server not running on %s", netAddr)
+	}
+
+	t.Parallel()
+
+	for _, test := range tests {
+		test := test
+		t.Run("default", func(t *testing.T) {
+			t.Parallel()
+
+			var buf [8]byte
+			if _, err := rand.Read(buf[:]); err != nil {
+				t.Fatal(err)
+			}
+			tableName := fmt.Sprintf("test_%x", buf[:])
+
+			db, err := sql.Open("mysql", dsn)
+			if err != nil {
+				t.Fatalf("error connecting: %s", err.Error())
+			}
+			t.Cleanup(func() {
+				db.Exec("DROP TABLE IF EXISTS " + tableName)
+				db.Close()
+			})
+
+			db.Exec("DROP TABLE IF EXISTS " + tableName)
+			dbt := &DBTest{t, db}
+			test(dbt, tableName)
+			dbt.db.Exec("DROP TABLE IF EXISTS test")
+		})
+
+		dsn2 := dsn + "&interpolateParams=true"
+		if _, err := ParseDSN(dsn2); err != errInvalidDSNUnsafeCollation {
+			t.Run("interpolateParams", func(t *testing.T) {
+				t.Parallel()
+
+				var buf [8]byte
+				if _, err := rand.Read(buf[:]); err != nil {
+					t.Fatal(err)
+				}
+				tableName := fmt.Sprintf("test_%x", buf[:])
```

Comment:
There is a non-zero risk of having two tests started with the same table name.

It would be better to use the index of the iteration of the `for` loop and a hash of the test name from which runTestsParallel is called.